packages/tools/src/eventDispatchers/shared/getToolsWithModesForTouchEvent.ts

```diff
@@ -0,0 +1,54 @@
+import { ToolGroupManager } from '../../store';
+import { ToolModes } from '../../enums';
+import { EventTypes } from '../../types';
+
+type ModesFilter = Array<ToolModes>;
+
+/**
+ * Given the normalized touch event and a filter of modes,
+ * find all the tools on the element that are in one of the specified modes.
+ * If the evtButton is specified, only tools with a matching binding will be returned.
+ * @param evt - The normalized touchStart event.
+ * @param modesFilter - An array of entries from the `ToolModes` enum.
+ */
+export default function getToolsWithModesForTouchEvent(
+  evt: EventTypes.NormalizedTouchEventType,
+  modesFilter: ModesFilter,
+  numTouchPoints?: number
+) {
+  const { renderingEngineId, viewportId } = evt.detail;
+  const toolGroup = ToolGroupManager.getToolGroupForViewport(
+    viewportId,
+    renderingEngineId
+  );
+
+  if (!toolGroup) {
+    return [];
+  }
+
+  const enabledTools = [];
+
+  const toolGroupToolNames = Object.keys(toolGroup.toolOptions);
+
+  for (let j = 0; j < toolGroupToolNames.length; j++) {
+    const toolName = toolGroupToolNames[j];
+    const tool = toolGroup.toolOptions[toolName];
+
+    const correctBinding =
+      numTouchPoints != null &&
+      tool.bindings.length &&
+      tool.bindings.some(
+        (binding) => binding.numTouchPoints === numTouchPoints
+      );
```

Comment:
For instance here, I expected to see the `numTouchPoints ===1 and mouse` thing in the other file

Reply:
In this case the behavior is to only return tools which match numTouchPoints, or if there are no touchpoints we include it in the list of tools. The numTouchPoints is analogous to mouse button.

https://github.com/cornerstonejs/cornerstone3D-beta/blob/87db14afede7def4588bb17f28d66963086e9410/packages/tools/src/eventDispatchers/shared/getToolsWithModesForMouseEvent.ts#L39-L49

The mouse primary logic would get superseeded by the !numTouchPoints